components/root/root.jsx

```diff
@@ -118,15 +122,19 @@ export default class Root extends React.PureComponent {
                 console.log('detected logout from a different tab'); //eslint-disable-line no-console
                 GlobalActions.emitUserLoggedOutEvent('/', false, false);
             }
+            if (e.originalEvent.key === StoragePrefixes.LOGIN && isNewLocalStorageEvent(e)) {
+                const isLoggedIn = getCurrentUser(store.getState());
 
-            if (e.originalEvent.key === StoragePrefixes.LOGIN && e.originalEvent.storageArea === localStorage && e.originalEvent.newValue) {
                 // make sure it isn't this tab that is sending the logout signal (only necessary for IE11)
-                if (BrowserStore.isSignallingLogin(e.originalEvent.newValue)) {
+                if (BrowserStore.isSignallingLogin(e.originalEvent.newValue) || isLoggedIn) {
                     return;
                 }
 
-                console.log('detected login from a different tab'); //eslint-disable-line no-console
-                location.reload();
+                // detected login from a different tab
+                function onVisibilityChange() {
+                    location.reload();
+                }
+                document.addEventListener('visibilitychange', onVisibilityChange, false);
```

Comment:
Why change this to only trigger when the window comes back into view?

Reply:
This was written in a comment on Jira: https://mattermost.atlassian.net/browse/MM-27861?focusedCommentId=96638

Reply:
Oh right, I forgot about that discussion. That works for me then.